src/theme/palette.ts

```diff
@@ -80,27 +84,30 @@ export const mainTypes = [
   'link',
 ] as const;
 
-export type flatPalette = Record<typeof flatColors[number], generatedColorShades>;
+export type FlatPalette = Record<typeof flatColors[number], GeneratedColorShades>;
 
-export type palePalette = Record<typeof paleColors[number], string>;
+export type PalePalette = Record<typeof paleColors[number], string>;
 
-export type generatedColorShades = Record<typeof colorShades[number], string>;
+export type GeneratedColorShades = Record<typeof colorShades[number], string>;
 
 /**
  * Palette is end output of what is produced and exported to the client projects
  **/
 export type Palette = {
+  flat: FlatPalette;
+  neutral: NeutralPaletteConfig;
+  gradient: GradientPaletteConfig;
+
+  /** @TODO remove all these when all components are revisited for v5 */
   text: {
-    primary: generatedColorShades;
-    secondary: generatedColorShades;
-    light: generatedColorShades;
+    primary: GeneratedColorShades;
+    secondary: GeneratedColorShades;
+    light: GeneratedColorShades;
   };
-  flat: flatPalette;
-  pale: palePalette;
-
+  pale: PalePalette;
```

Comment:
also a TODO here for removal

Reply:
the comment on line 101 refers to all the lines until line 110